packages/material-ui/src/Rating/Rating.js

```diff
@@ -340,15 +332,14 @@ const Rating = React.forwardRef(function Rating(props, ref) {
 
     return (
       <React.Fragment key={state.value}>
-        <label className={classes.label} htmlFor={id} {...labelProps}>
+        <label className={classes.label} {...labelProps}>
           {container}
           <span className={classes.visuallyHidden}>{getLabelText(state.value)}</span>
         </label>
         <input
           onFocus={handleFocus}
           onBlur={handleBlur}
-          onChange={handleChange}
-          onClick={handleClear}
+          readOnly
```

Comment:
The `readOnly` was needed to avoid needing an `onChange` handler since this is now controlled based on where the mouse is when clicked, rather than having the hover state and click events calculated separately